benchmarks/corrections/test_corrset.py

```diff
@@ -41,3 +42,102 @@ def test_tileshape_adjustment_bench(benchmark, base_shape, excluded_coords):
     print("Base shape", base_shape)
     print("Excluded coords", excluded_coords)
     print("Adjusted", adjusted)
+
+
+def _make_data(nav_dims, sig_dims):
+    data = np.linspace(
+        start=5, stop=30, num=np.prod(nav_dims) * np.prod(sig_dims), dtype=np.float32
+    )
+    return data.reshape(nav_dims + sig_dims)
+
+
+def _generate_exclude_pixels(sig_dims, num_excluded):
+    '''
+    Generate a list of excluded pixels that
+    can be reconstructed faithfully from their neighbors
+    in a linear gradient dataset
+    '''
```

Comment:
This is copy-pasta'd from the tests - possibly share this between benchmarks and tests.